packages/gcf-utils/src/gcf-utils.ts

```diff
@@ -45,6 +47,48 @@ interface EnqueueTaskParams {
   name: string;
 }
 
+interface LogFn {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (msg: string, ...args: any[]): void;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (obj: object, msg?: string, ...args: any[]): void;
+}
+
+/**
+ * A logger standardized logger for Google Cloud Functions
+ */
+export interface GCFLogger {
+  [key: string]: Function;
```

Comment:
Do we need this? It looks like it's used for tests to access the log functions by grabbing the function via js object properties.

Is there a better way to access that in TS? @bcoe @JustinBeckwith

Reply:
This is just for testing, specifically here to do:

```typescript

export const logLevels: {[index: string]: number} = {
  trace: 10,
  debug: 20,
  info: 30,
  metric: 30,
  warn: 40,
  error: 50,
};

for (let level of Object.keys(logLevels)) {
  logger[level](`log a statement of severity ${level$}`);
}
```

This makes testing a lot more concise, but if there's a workaround for this I'm happy to remove the mapping

Reply:
I would really avoid doing that here.  Adding `[key: string]: Function;` will really water down the effectiveness of the type safety provided, and make it possible to do all kinds of stuff that would fail at runtime, but the compiler would happily allow.  

If the only reason you want this is for testing, I would do a cast in the test itself to something looser, while not loosening the types for library consumers.

Reply:
Makes sense - I'll try to see if I can make it work without the `[key: string]: Function` since it's being used for casting Pino as well other than just a testing help. I think I have some workarounds that might do.